--- a/grammar.js
+++ b/grammar.js
@@ -91,7 +91,7 @@
     // 0/texture = ExtResource( 2 )
     _properties: ($) => repeat1($.property),
     property: ($) => seq($.path, "=", $._value),
-    path: ($) => /[a-zA-Z_0-9][a-zA-Z_:/0-9]*/,
+    path: ($) => /[a-zA-Z_0-9][a-zA-Z_:/0-9.]*/,
 
     // -----------------------------------------------------------------------------
     // -                                 Data Structs                              -
@@ -99,9 +99,9 @@
 
     pair: ($) => seq($._value, ":", $._value),
 
-    dictionary: ($) => seq("{", optional(commaSep1($.pair)), "}"),
+    dictionary: ($) => seq("{", optional(seq(commaSep1($.pair), optional(","))), "}"),
 
-    array: ($) => seq("[", optional(commaSep1($._value)), "]"),
+    array: ($) => seq("[", optional(seq(commaSep1($._value), optional(","))), "]"),
 
     // -----------------------------------------------------------------------------
     // -                                  Constructor                              -
