diff --git a/common/define-grammar.js b/common/define-grammar.js
index efb0b76..2a603cb 100644
--- a/common/define-grammar.js
+++ b/common/define-grammar.js
@@ -322,8 +322,10 @@ module.exports = function defineGrammar(dialect) {
         seq(
           'export',
           'type',
-          $.export_clause,
-          optional($._from_clause),
+          choice(
+            seq(choice("*", $.namespace_export), $._from_clause),
+            seq($.export_clause, optional($._from_clause)),
+          ),
           $._semicolon,
         ),
         seq('export', '=', $.expression, $._semicolon),
@@ -726,6 +728,4 @@ module.exports = function defineGrammar(dialect) {
         $.readonly_type,
         $.constructor_type,
         $.infer_type,
-        prec(-1, alias($._type_query_member_expression_in_type_annotation, $.member_expression)),
-        prec(-1, alias($._type_query_call_expression_in_type_annotation, $.call_expression)),
       ),
@@ -764,4 +764,6 @@ module.exports = function defineGrammar(dialect) {
       primary_type: $ => choice(
+        prec(-1, alias($._type_query_member_expression_in_type_annotation, $.member_expression)),
+        prec(-1, alias($._type_query_call_expression_in_type_annotation, $.call_expression)),
         $.parenthesized_type,
         $.predefined_type,
         $._type_identifier,
